server/app/views/admin/programs/ProgramIndexView.java

```diff
@@ -89,24 +96,55 @@ public Content render(
                                     programs.getDraftProgramDefinition(name),
                                     request,
                                     profile))))
-            .with(renderDownloadExportCsvButton());
+            .with(demographicsCsvModal.getButton());
 
     HtmlBundle htmlBundle =
         layout
             .getBundle()
             .setTitle(pageTitle)
             .addMainContent(contentDiv)
-            .addModals(publishAllModal)
+            .addModals(publishAllModal, demographicsCsvModal)
             .addFooterScripts(layout.viewUtils.makeLocalJsTag("admin_programs"));
     return layout.renderCentered(htmlBundle);
   }
 
-  private ContainerTag renderDownloadExportCsvButton() {
-    return new LinkElement()
-        .setId("download-export-csv-button")
-        .setHref(routes.AdminApplicationController.downloadDemographics().url())
-        .setText("Download Exported Data (CSV)")
-        .asButton();
+  private Modal renderDemographicsCsvModal() {
+    String modalId = "download-demographics-csv-modal";
+    String downloadActionText = "Download Exported Data (CSV)";
+
+    ContainerTag downloadDemographicCsvModalContent =
+        div()
+            .withClasses(Styles.PX_8)
+            .with(
+                form()
+                    .withMethod("GET")
+                    .withAction(routes.AdminApplicationController.downloadDemographics().url())
+                    .with(
+                        p("This will download all applications for all programs and can take"
```

Comment:
this is a bit hard to read. Rephrase into something like:
"This will download all applications for all programs. Use filters below to select date range for the data. Warning: if you select a large date range or leave it blank it might take up to (10? 30?) minutes to export data."